src/interpreter/ByteCode.h

```diff
@@ -256,6 +256,8 @@ class ByteCode {
 
     Opcode opcode() const;
     size_t getSize();
+    const std::string getName() const { return "MISSING NAME"; };
+    void printExtra(size_t pos) { return; };
```

Comment:
Are these functions used?

Reply:
I wanted to ensure that every child class of the ByteCode class has these functions. My original aim was to ensure that it won't cause any (runtime) errors if it's not declared in a child class. But now, I've been thinking about it. Maybe it would be better if it fails, if it's not declared? It seems better to me, that it forces the programmer to write these functions.

Should I remove them from ByteCode class?

Reply:
These print functions are used only for `debug` mode to provide debugging information to programmers.
Moreover, these functions are no longer defined as virtual functions of `ByteCode` class.
So, they cannot force child class to define its own function and they are even not invoked at all.
To sum up, IMHO its better to remove them.

Reply:
Yes, but the macros at `dumpByteCode()` in Module.cpp do demand them. Without declaring them, the compiler throws errors.

Here:
``` C++
    while (idx < m_byteCode.size()) {
        ByteCode* code = reinterpret_cast<ByteCode*>(&m_byteCode[idx]);
        printf("| %15zu |", idx);
        switch (code->opcode()) {
#define GENERATE_BYTECODE_CASE(name, ...)                                 \
    case ByteCode::name##Opcode:                                          \
        printf("%5zu |", static_cast<name*>(code)->getSize());            \
        printf(" %-15s | ", static_cast<name*>(code)->getName().c_str()); \
        static_cast<name*>(code)->printExtra(idx);                        \
        break;

            FOR_EACH_BYTECODE(GENERATE_BYTECODE_CASE);
#undef GENERATE_BYTECODE_CASE
        default:
            ASSERT_NOT_REACHED();
            break;
        }
```

So, I'd remove the declarations from the parent class to ensure that they have to be declared in the child classes (or at least in the classes that are used in `dumpByteCode()`).

Meanwhile co-worker Ádám came up with the idea, that `getName()` function can be replaced with the macro variable. Thus, only one function is enought, the `printExtra()`. I think it's a great idea, because it makes the code more straightforward and gives less change to make mistakes.